tests/pytests/scenarios/failover/multimaster/test_failover_master.py

```diff
@@ -0,0 +1,306 @@
+    salt_mm_failover_minion_1,
+    salt_mm_failover_minion_2,
+    mm_failover_master_1_salt_cli,
+    mm_failover_master_2_salt_cli,
+):
+    """
+    Test then when the first master is stopped, connected minions failover to the second master.
+    """
+    # Get all the minions connected to salt_mm_failover_master_1
+    master_1_minions = _run_echo_for_all_possibilities(
+        [mm_failover_master_1_salt_cli],
+        [salt_mm_failover_minion_1, salt_mm_failover_minion_2],
+    )
+    event_patterns = [
+        (minion.id, "salt/minion/{}/start".format(minion.id))
+        for minion in master_1_minions
+    ]
+
+    # breakpoint()
+    with salt_mm_failover_master_1.stopped():
+        start_time = time.time()
+        # We need to wait for them to realize that the master is not alive
+        # At this point, only the first minion will need to change masters
+        if event_patterns:
+            event_listener.wait_for_events(
+                event_patterns,
+                timeout=salt_mm_failover_minion_1.config["master_alive_interval"] * 2,
+                after_time=start_time,
+            )
+
+        _run_echo_for_all_possibilities(
+            [mm_failover_master_1_salt_cli, mm_failover_master_2_salt_cli],
+            [salt_mm_failover_minion_1, salt_mm_failover_minion_2],
+        )
+
+        # We are getting the return events associated with each minion
+        minion_1_pattern = "salt/job/*/ret/{}".format(salt_mm_failover_minion_1.id)
+        minion_2_pattern = "salt/job/*/ret/{}".format(salt_mm_failover_minion_2.id)
+
+        # Make sure nothing returned to the first (stopped) master
+        assert not event_listener.get_events(
+            [(salt_mm_failover_master_1.id, minion_1_pattern)], after_time=start_time
+        )
+        assert not event_listener.get_events(
+            [(salt_mm_failover_master_1.id, minion_2_pattern)], after_time=start_time
+        )
+
+        minion_1_ret_events = event_listener.get_events(
+            [(salt_mm_failover_master_2.id, minion_1_pattern)], after_time=start_time
+        )
+        minion_2_ret_events = event_listener.get_events(
+            [(salt_mm_failover_master_2.id, minion_2_pattern)], after_time=start_time
+        )
+
+        # Each minion should only return to one master
+        assert len(minion_1_ret_events) == 1
+        assert len(minion_2_ret_events) == 1
+
+
+def test_minion_reconnection_against_one_live_master(
+    event_listener,
+    salt_mm_failover_master_1,
+    salt_mm_failover_master_2,
+    salt_mm_failover_minion_1,
+    salt_mm_failover_minion_2,
+    mm_failover_master_1_salt_cli,
+    mm_failover_master_2_salt_cli,
+):
+    """
+    Test that mininons reconnect to a live master.
+
+    To work well with salt factories, the minions will reconnect to the master the were connected to in conftest.py.
+    We should keep this test directly after `test_failover_to_second_master`, to ensure all minions are initially
```

Comment:
We want to prevent having to depend on another test. Can we add a pytest factory that ensures all services are up and running before each test, to ensure we don't need to depend on another test.

Reply:
Yes, that makes sense.  For some reason, I remember this comment actually not holding true anymore, so I will investigate and add a fix if it truly does depend on the previous test

Reply:
After checking it out, it seems I was remembering correctly.  There is no dependency on another test and the comment was outdated.